src/app/games/teeko/teeko.component.ts

```diff
@@ -0,0 +1,115 @@
+import { TeekoRules } from './TeekoRules';
+import { TeekoDropMove, TeekoMove, TeekoTranslateMove } from './TeekoMove';
+import { TeekoState } from './TeekoState';
+import { Component } from '@angular/core';
+import { MessageDisplayer } from 'src/app/services/MessageDisplayer';
+import { TeekoMinimax } from './TeekoMinimax';
+import { TeekoTutorial } from './TeekoTutorial';
+import { RectangularGameComponent } from 'src/app/components/game-components/rectangular-game-component/RectangularGameComponent';
+import { Player, PlayerOrNone } from 'src/app/jscaip/Player';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { Coord } from 'src/app/jscaip/Coord';
+import { MGPValidation } from 'src/app/utils/MGPValidation';
+import { RulesFailure } from 'src/app/jscaip/RulesFailure';
+
+@Component({
+    selector: 'app-teeko',
+    templateUrl: './teeko.component.html',
+    styleUrls: ['../../components/game-components/game-component/game-component.scss'],
+})
+
+export class TeekoComponent extends RectangularGameComponent<TeekoRules,
+                                                             TeekoMove,
+                                                             TeekoState,
+                                                             PlayerOrNone>
+{
+
+    public selected: MGPOptional<Coord> = MGPOptional.empty();
+    public last: MGPOptional<Coord> = MGPOptional.empty();
+    public moved: Coord[] = [];
+    public victory: Coord[] = [];
+
+    public constructor(messageDisplayer: MessageDisplayer) {
+        super(messageDisplayer);
+        this.rules = TeekoRules.get();
+        this.node = this.rules.getInitialNode();
+        this.encoder = TeekoMove.encoder;
+        this.tutorial = new TeekoTutorial().tutorial;
+        this.availableMinimaxes = [
+            new TeekoMinimax(),
+        ];
+    }
+    public updateBoard(): void {
+        this.board = this.node.gameState.board;
+    }
+    public override showLastMove(move: TeekoMove): void {
+        this.last = MGPOptional.of(this.rules.getLastCoord(move));
+        if (move instanceof TeekoTranslateMove) {
+            this.moved = [move.getStart(), move.getEnd()];
+        } else {
+            this.moved = [];
+        }
+        this.victory = this.rules.getVictoryCoord(this.getState());
+    }
+    public hideLastMove(): void {
+        this.last = MGPOptional.empty();
+        this.moved = [];
+        this.victory = [];
+    }
+    public override cancelMoveAttempt(): void {
+        this.selected = MGPOptional.empty();
+    }
+    public async onClick(x: number, y: number): Promise<MGPValidation> {
+        const clickValidity: MGPValidation = this.canUserPlay('#click_' + x + '_' + y);
+        if (clickValidity.isFailure()) {
+            return this.cancelMove(clickValidity.getReason());
+        }
+        const clickedCoord: Coord = new Coord(x, y);
+        if (this.getState().turn < 8) {
```

Comment:
les bonnes pratiques sur les nombres magiques demanderaient d'avoir `this.getState().isInDropPhase()` et le 8 qui apparait uniquement dans cette fonction